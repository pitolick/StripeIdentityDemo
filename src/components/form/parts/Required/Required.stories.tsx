import React from 'react'
import { Meta, StoryObj } from '@storybook/react'
import { Required } from './Required'

const meta: Meta<typeof Required> = {
  title: 'form/parts/Required',
  component: Required,
  tags: ['autodocs'],
}

export default meta
type Story = StoryObj<typeof Required>

/**
 * デフォルト（引数無しだとデフォルトで必須として出力）
 */
export const Default: Story = {}

/**
 * 必須
 */
export const Require: Story = {
  args: {
    required: true,
  },
}

/**
 * 任意
 */
export const Optional: Story = {
  args: {
    required: false,
  },
}
