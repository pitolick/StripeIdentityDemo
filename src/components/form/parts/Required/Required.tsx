/**
 * 必須任意項目
 * @module Required
 */
import React from 'react'

/**
 * 必須任意項目コンポーネントのプロパティ
 */
interface RequiredProps {
  /**
   * 必須項目か
   */
  required?: boolean
}

/**
 * 必須任意項目コンポーネント
 */
export const Required: React.FC<RequiredProps> = ({ required = true }) => {
  const requiredClass = 'text-xs text-center px-1.5 md:px-3 py-1 rounded-md whitespace-nowrap'
  return <>{required ? <span className={`${requiredClass} bg-red text-white`}>必須</span> : <span className={`${requiredClass} bg-gray-500 text-white`}>任意</span>}</>
}
