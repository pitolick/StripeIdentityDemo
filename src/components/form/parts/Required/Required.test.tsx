import React from 'react'
import { render } from '@testing-library/react'
import { Required } from './Required'

describe('Requiredコンポーネント', () => {
  it('必須項目の場合、"必須"と表示されること', () => {
    const { container } = render(<Required required={true} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('任意項目の場合、"任意"と表示されること', () => {
    const { container } = render(<Required required={false} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('requiredが未入力の場合、"必須"と表示されること', () => {
    const { container } = render(<Required required={undefined} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
