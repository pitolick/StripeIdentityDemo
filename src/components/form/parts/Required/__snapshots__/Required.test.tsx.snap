// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Requiredコンポーネント requiredが未入力の場合、"必須"と表示されること 1`] = `
<span
  class="text-xs text-center px-1.5 md:px-3 py-1 rounded-md whitespace-nowrap bg-red text-white"
>
  必須
</span>
`;

exports[`Requiredコンポーネント 任意項目の場合、"任意"と表示されること 1`] = `
<span
  class="text-xs text-center px-1.5 md:px-3 py-1 rounded-md whitespace-nowrap bg-gray-500 text-white"
>
  任意
</span>
`;

exports[`Requiredコンポーネント 必須項目の場合、"必須"と表示されること 1`] = `
<span
  class="text-xs text-center px-1.5 md:px-3 py-1 rounded-md whitespace-nowrap bg-red text-white"
>
  必須
</span>
`;
